packages/vitest/src/runtime/mocker.ts

```diff
@@ -209,10 +209,21 @@ export class VitestMocker {
       const isModule = containerType === 'Module' || !!container.__esModule
       for (const { key: property, descriptor } of getAllMockableProperties(container)) {
         // Modules define their exports as getters. We want to process those.
-        if (!isModule) {
-          // TODO: Mock getters/setters somehow?
-          if (descriptor.get || descriptor.set)
-            continue
+        if (!isModule && descriptor.get) {
+          try {
+            const newDescriptor: PropertyDescriptor = {
+              enumerable: true,
+              configurable: true,
+              get: spyModule.fn(),
+            }
+            if (descriptor.set)
+              newDescriptor.set = spyModule.fn()
+            Object.defineProperty(newContainer, property, newDescriptor)
+          }
+          catch (error) {
+            // Ignore errors, just move on to the next prop.
+          }
+          continue
```

Comment:
isn't this `continue` redundant? perhaps you could move it into the `catch`

Reply:
No, if we encounter a getter we don't want to do any of the other mocking logic afterwards. The `continue` here says "we're done with this prop, `continue` on to the next one". The `catch` is just to allow graceful handling of trying to overwrite non-configurable props.